test/python/circuit/library/test_mcmt.py

```diff
@@ -268,18 +269,30 @@ def test_vchain_open_control(self):
             )
         )
 
-    def test_mcmt_x_gate(self):
+    @combine(num_ctrl=range(1, 6), num_targ=range(2, 6))
```

Comment:
This goes up to 12 qubit circuits, right? It seems a bit overkill to compute the full 2^12 x 2^12 sized matrix, maybe we can reduce this a bit? E.g. 
```suggestion
    @combine(num_ctrl=range(1, 3), num_targ=range(2, 4))
```